### ISSUE TITLE
Day label is drifting on the left when calendar range is getting smaller

### ISSUE THREAD

**Issue by Winbee** [apache/echarts#11508]
### Version
4.4.0

### Reproduction link
[https://www.echartsjs.com/examples/en/editor.html?c=calendar-horizontal](https://www.echartsjs.com/examples/en/editor.html?c=calendar-horizontal)

### Steps to reproduce
- Open the example https://www.echartsjs.com/examples/en/editor.html?c=calendar-horizontal
- Decrease the range of the first calendar like that:
range: ['2017-01', '2017-04'],
range: ['2017-01', '2017-03'],
range: ['2017-01', '2017-02'],

### What is expected?
The day label should remain at the same position.

### What is actually happening?
The day label is moving further and and further on the left when we decrease the range.

**Comment by Winbee** [comment]
range: ['2017-01', '2017-04'],
<img width="189" alt="Screenshot 2019-10-29 at 8 21 20" src="https://user-images.githubusercontent.com/7758451/67745968-ca85a180-fa25-11e9-9744-bda9e65fba4f.png">

range: ['2017-01', '2017-03'],
<img width="190" alt="Screenshot 2019-10-29 at 8 24 47" src="https://user-images.githubusercontent.com/7758451/67745977-ce192880-fa25-11e9-9723-2a73aa01867e.png">

range: ['2017-01', '2017-02'],
<img width="205" alt="Screenshot 2019-10-29 at 8 25 01" src="https://user-images.githubusercontent.com/7758451/67745982-d1141900-fa25-11e9-9bfe-fbf2723b7450.png">

**Comment by yufeng04** [comment]
It seems to be a bug!

**Comment by Nick22nd** [comment]
The issue as I tracking：

If orient is horizontal and range changes, the cell will automatically adjust its width.
But the weekly text margin is always equalling 50% width of cellSize.

I tried to fix it by making daylabel margin always be the cell's height . 
It seems that the issue will be fixed. But I have 2 problem:
1. I don't know whether it will cause another issue.
2. I only changed 1 line code ,should I make a PR?
